src/plugins/links/public/embeddable/links_embeddable.tsx

```diff
@@ -6,155 +6,271 @@
+          defaultPanelDescription,
+          isEditingEnabled: () => Boolean(error$.value === undefined),
+          libraryId$: savedObjectId$,
+          getTypeDisplayName: () => DISPLAY_NAME,
+          getByValueRuntimeSnapshot: () => {
+            const snapshot = api.snapshotRuntimeState();
+            delete snapshot.savedObjectId;
+            return snapshot;
+          },
+          serializeState: async (): Promise<SerializedPanelState<LinksSerializedState>> => {
+            if (savedObjectId$.value !== undefined) {
+              const linksByReferenceState: LinksByReferenceSerializedState = {
+                savedObjectId: savedObjectId$.value,
+                ...serializeTitles(),
+              };
+              return { rawState: linksByReferenceState, references: [] };
+            }
+            const runtimeState = api.snapshotRuntimeState();
+            const { attributes, references } = serializeLinksAttributes(runtimeState);
+            const linksByValueState: LinksByValueSerializedState = {
+              attributes,
+              ...serializeTitles(),
+            };
+            return { rawState: linksByValueState, references };
+          },
+          saveToLibrary: async (newTitle: string) => {
+            defaultPanelTitle.next(newTitle);
+            const runtimeState = api.snapshotRuntimeState();
+            const { attributes, references } = serializeLinksAttributes(runtimeState);
+            const {
+              item: { id },
+            } = await linksClient.create({
+              data: {
+                ...attributes,
+                title: newTitle,
+              },
+              options: { references },
+            });
+            savedObjectId$.next(id);
+            return id;
+          },
+          checkForDuplicateTitle: async (
+            newTitle: string,
+            isTitleDuplicateConfirmed: boolean,
+            onTitleDuplicate: () => void
+          ) => {
+            await checkForDuplicateTitle({
+              title: newTitle,
+              copyOnSave: false,
+              lastSavedTitle: '',
+              isTitleDuplicateConfirmed,
+              onTitleDuplicate,
+            });
+          },
+          unlinkFromLibrary: () => {
+            savedObjectId$.next(undefined);
+          },
+          onEdit: async () => {
+            try {
+              const { openEditorFlyout } = await import('../editor/open_editor_flyout');
+              const newState = await openEditorFlyout({
+                initialState: api.snapshotRuntimeState(),
+                parentDashboard: parentApi,
+              });
+              if (newState) {
+                links$.next(newState.links);
+                layout$.next(newState.layout);
+                defaultPanelTitle.next(newState.defaultPanelTitle);
+                defaultPanelDescription.next(newState.defaultPanelDescription);
+                savedObjectId$.next(newState.savedObjectId);
+              }
```

Comment:
As expected, this closes https://github.com/elastic/kibana/issues/186044 🎉  🎉  🎉